### PR TITLE
Feature new sdk demo mode

### LINKED ISSUE #1288: Add demo mode
- [x] Add demo button on Login.
- [x] Demo pull.
- [x] Demo push.
- [x] Move executeDemo to normal launchpull in progressactivity and show progressactivity.
- [x] Create DB.
- [x] Create CSV.
- [x] Copy DB from assets.
- [x] Solve bug when DB is copy from assets: The app needs a restart to use the DB. 
- [x] Populate from CSV.

### PR DESCRIPTION
**Pull request by idelcano**
Closes #1288

### CODE REVIEW

File: app/src/main/java/org/eyeseetea/malariacare/data/database/local/PullLocalSDKDataSource.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+import org.eyeseetea.malariacare.data.IPullSourceCallback;
+import org.eyeseetea.malariacare.data.database.AppDatabase;
+import org.eyeseetea.malariacare.data.database.datasources.ConversionLocalDataSource;
+import org.eyeseetea.malariacare.data.database.utils.PopulateDB;
+import org.eyeseetea.malariacare.utils.FileIOUtils;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+public class PullLocalSDKDataSource {
```
Comment (xurxodev) [inline app/src/main/java/org/eyeseetea/malariacare/data/database/local/PullLocalSDKDataSource.java]: @idelcano Now exists local and remote PullDataSource, should exists interface for this two implementations.
Can you rename PullLocalSDKDataSource to PullLocalDataSource and move to database/datasource package?

File: app/src/main/java/org/eyeseetea/malariacare/data/database/iomodules/dhis/importer/PullController.java [hunk 2]
```diff
@@ -178,4 +197,13 @@ public void cancel() {
     public boolean isPullActive() {
         return PULL_IS_ACTIVE;
     }
+
+    public class LocalAsyncPull extends AsyncTask<Void, Void, Void> {
```
Comment (xurxodev) [inline app/src/main/java/org/eyeseetea/malariacare/data/database/iomodules/dhis/importer/PullController.java]: @idelcano PullLocalDataSource receive a callback as parameter then better that inside this class execute in background and not from PullController equal to PullRemoteDataSource

File: app/src/main/java/org/eyeseetea/malariacare/data/database/iomodules/dhis/importer/PullController.java [hunk 3]
```diff
@@ -93,16 +88,23 @@ public void postFinish() {
 
     @Override
     public void pull(final PullFilters filters, final IPullControllerCallback callback) {
```
Comment (xurxodev) [inline app/src/main/java/org/eyeseetea/malariacare/data/database/iomodules/dhis/importer/PullController.java]: @idelcano can you refactor this method extracting logic to new methods ExexuteDemoPull and ExecuteRemotePull for have the code more readable as UserAccountRepository?

File: app/src/eds/java/org/eyeseetea/malariacare/strategies/LoginActivityStrategy.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+            @Override
+            public void onClick(View v) {
+
+                Credentials demoCrededentials = Credentials.createDemoCredentials();
+
+                loginActivity.mLoginUseCase.execute(demoCrededentials,
+                        new LoginUseCase.Callback() {
+                            @Override
+                            public void onLoginSuccess() {
+                                loginActivity.onSuccess();
```
Comment (xurxodev) [inline app/src/eds/java/org/eyeseetea/malariacare/strategies/LoginActivityStrategy.java]: @idelcano I think that after demo login we should not navigate to ProgressActvity, we should execute demo pull as in pictureapp

**Comment by idelcano** [comment]
@xurxodev 
i updated this PR.
i reviewed your comments
And i solved an error with the values id.

File: app/src/main/java/org/eyeseetea/malariacare/data/database/iomodules/dhis/importer/PullController.java (hunk 3 above)
Comment (idelcano) [inline app/src/main/java/org/eyeseetea/malariacare/data/database/iomodules/dhis/importer/PullController.java]: I moved the pull demo to the LocalPullController as we talk.

File: app/src/eds/java/org/eyeseetea/malariacare/strategies/LoginActivityStrategy.java [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+import org.eyeseetea.malariacare.data.database.model.User;
+import org.eyeseetea.malariacare.domain.entity.Credentials;
+import org.eyeseetea.malariacare.domain.usecase.LoadUserAndCredentialsUseCase;
+import org.eyeseetea.malariacare.domain.usecase.LoginUseCase;
+import org.eyeseetea.malariacare.domain.usecase.pull.PullFilters;
+import org.eyeseetea.malariacare.domain.usecase.pull.PullStep;
+import org.eyeseetea.malariacare.domain.usecase.pull.PullUseCase;
+import org.hisp.dhis.client.sdk.ui.views.FontButton;
+
+public class LoginActivityStrategy extends ALoginActivityStrategy {
```
Comment (xurxodev) [inline app/src/eds/java/org/eyeseetea/malariacare/strategies/LoginActivityStrategy.java]: LoginActivityStrategy is duplicate in eds and hnqis, can you refactor for have only one in main and delete ALoginActivityStrategy?

**Comment by idelcano** [comment]
@xurxodev  it's done !

**Pull request merged**